path/resolver.go

```diff
@@ -121,9 +123,12 @@ func (s *Resolver) ResolveLinks(ndd *merkledag.Node, names []string) (
 
 		if nlink.Node == nil {
 			// fetch object for link and assign to nd
-			ctx, cancel := context.WithTimeout(context.TODO(), time.Minute)
-			defer cancel()
-			nd, err = s.DAG.Get(ctx, next)
+			/*
+				TODO(cryptix): we wrapped a new context for each name iteration
+				is this correct?
+				moved it out of the loop for now
+			*/
```

Comment:
i think this used to be right. we should probably give longer names more time. example: if resolution takes ~10s per hop, 1min is not going to cut it for paths larger than 6. (resolution should be much faster, sure, but remember that resolution might go across the routing system).

in the end, this probably should come from the context coming in, and not have a timeout internally.